### PR TITLE
docs(api): improve section on nvim_set_hl

### PR DESCRIPTION
**Pull request by adrian5**

### CODE REVIEW

**Comment by clason** [comment]
Thank you, but these docs are autogenerated from the docstring in the code -- so any improvements must be made to the latter, or they will just get overwritten again.

(Personally, I don't actually see much improvement in your changes -- the markup is needed for `:help`.)

**Comment by adrian5** [comment]
I see, then I'll remove the first part. Is the commit still `docs(api)` in that case?

> Personally, I don't actually see much improvement in your changes

Well, aside from having multiple typos in that tiny section, I think the formatting made it hard to read (compared to neighboring functions). The `cterm` explanation also seemed more confusing than helpful.

> the markup is needed for `:help`

You mean `ctermfg`: vs plain cterm in the parameter list? I removed those to match the formatting used in the surrounding functions, as it looked more readable that way. But if it's important for functionality, I can add it back in.

**Comment by clason** [comment]
> I see, then I'll remove the first part. Is the commit still docs(api) in that case?

You don't have to remove it, but it must be consistent. Yes, it's still about docs.

> Well, aside from having multiple typos in that tiny section, I think the formatting made it hard to read (compared to neighboring functions). The cterm explanation also seemed more confusing than helpful.

Sure, that's an an improvement -- although I don't quite agree about the changes. I do think the reference to `hi default` is useful here, and the new wording for `cterm` could be misunderstood: "sharing" means a bidirectional link, so setting `cterm` would change `gui` as well. Maybe "attributes default to those set for `gui` if not set"?

File: runtime/doc/api.txt [hunk 1]
```diff
@@ -1547,20 +1547,19 @@ nvim_set_hl({ns_id}, {name}, {*val})                           *nvim_set_hl()*
                 Set a highlight group.
 
                 Parameters: ~
-                    {ns_id}  number of namespace for this highlight. Use value
+                    {ns_id}  Number of namespace for this highlight. Use value
                              0 to set a highlight group in the global (
                              `:highlight` ) namespace.
```
Comment (gpanders) [inline runtime/doc/api.txt:1552]:
```suggestion
                    {ns_id}  Namespace id for this highlight |nvim_create_namespace()|. 
                             Use 0 to set a highlight group globally |:highlight|.
```

File: runtime/doc/api.txt [hunk 2]
```diff
@@ -1547,20 +1547,19 @@ nvim_set_hl({ns_id}, {name}, {*val})                           *nvim_set_hl()*
...
-                             keys are also recognized: `default` : don't
-                             override existing definition, like `hi default`
-                             `ctermfg` : sets foreground of cterm color
-                             `ctermbg` : sets background of cterm color
-                             `cterm` : cterm attribute map. sets attributed
-                             for cterm colors. similer to `hi cterm` Note: by
-                             default cterm attributes are same as attributes
-                             of gui color
+                    {name}   Highlight group name, e.g. "ErrorMsg"
+                    {val}    Highlight definition map, like |synIDattr|. In
```
Comment (gpanders) [inline runtime/doc/api.txt:1554]:
```suggestion
                    {val}    Highlight definition map, like |synIDattr()|. In
```

File: runtime/doc/api.txt [hunk 3]
```diff
@@ -1547,20 +1547,19 @@ nvim_set_hl({ns_id}, {name}, {*val})                           *nvim_set_hl()*
...
+                             addition, the following keys are recognized:
+                             • default: Don't override existing definition
```
Comment (gpanders) [inline runtime/doc/api.txt:1556]:
```suggestion
                             • default: Don't override existing definition |:hi-default|
```

File: runtime/doc/api.txt [hunk 4]
```diff
@@ -1547,20 +1547,19 @@ nvim_set_hl({ns_id}, {name}, {*val})                           *nvim_set_hl()*
...
+                             • ctermfg: Sets foreground of cterm color
+                             • ctermbg: Sets background of cterm color
```
Comment (gpanders) [inline runtime/doc/api.txt:1558]:
```suggestion
                             • ctermfg: Sets foreground of cterm color |highlight-ctermfg|
                             • ctermbg: Sets background of cterm color |highlight-ctermbg|
```

**Review by gpanders — CHANGES_REQUESTED** [review]
Thanks for the PR! Doc improvements are always welcome.

I added a few of my own suggestions.

**Comment by adrian5** [comment]
> `cterm`  : cterm attribute map. sets attributed for
>                cterm colors. similer to `hi cterm`

I meant the first part being "confusing", doesn't `cterm` solely affect text properties such as boldness?

As for the second part, I didn't understand what it was pointing at. `hi cterm` alone is not a working command, so was this about how it's used similar to `cterm` in e.g. `hi GroupName cterm=…`?

> I do think the reference to `hi default` is useful here

Right, my bad. I think I got mixed up with `hi cterm` after purging that.

> and the new wording for cterm could be misunderstood: "sharing" means a bidirectional link, so setting cterm would change gui as well. Maybe "attributes default to those set for gui if not set"?

Agreed, I'm not happy with that wording either. Your sentence seems unambiguous.

**Comment by clason** [comment]
> As for the second part, I didn't understand what it was pointing at. hi cterm alone is not a working command, so was this about how it's similar in use to cterm in e.g. hi GroupName cterm=…?

Agreed, that's not as clear as it should be. Yes, this is exactly about the `cterm=...` attributes.

**Comment by adrian5** [comment]
@gpanders Those are really good. I'm not familiar with the GitHub UI, can I just merge these into my commit via the buttons? Do they need to be squashed afterwards?

**Comment by clason** [comment]
Yes, and yes. (Commit lint won't be happy otherwise.)

**Comment by adrian5** [comment]
Ok, great. I'll take of that later when I get home.

**Comment by adrian5** [comment]
I've merged the suggested changes and transferred them to the source docstring. I think this is a real improvement for the reader. I hope the changes are in order now.

**Review by clason — APPROVED** [review]

**Review by gpanders — APPROVED** [review]

**Pull request merged**